expression/constraint_propagation.go

```diff
@@ -164,3 +164,9 @@ func ruleColumnEQConst(ctx sessionctx.Context, i, j int, exprs *exprSet) {
 		}
 	}
 }
+
+// RemoveDupExprs removes identical exprs using exprSet.
+func RemoveDupExprs(exprs []Expression) []Expression {
```

Comment:
Will be some expressions that although have same hash code, but they will have side effects so we cannot just evaluate them once?